src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -416,6 +417,13 @@ bool Condition::Meets(ConditionSourceInfo& sourceInfo)
                 condMeets = creature->GetCreatureTemplate()->type == ConditionValue1;
             break;
         }
+        case CONDITION_REALM_ACHIEVEMENT:
+        {
+            AchievementEntry const* achievement = sAchievementMgr->GetAchievement(ConditionValue1);
+            if (sAchievementMgr->IsRealmCompleted(achievement, ConditionValue2))
```

Comment:
Since instanceId is 2nd argument to IsRealmCompleted, do not pass ConditionValue2 for it, just use std::numeric_limits<uint32>::max()